public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -98,7 +98,14 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
 
     for (const target of filteredTargets) {
       if (options.app === CoreApp.Explore) {
-        subQueries.push(this.runInstantQuery(target, options, filteredTargets.length));
+        subQueries.push(
+          this.runInstantQuery(target, options, filteredTargets.length).pipe(
+            map(response => {
+              // Let rangeQuery finish the loading
+              return { ...response, state: LoadingState.Loading };
```

Comment:
What if instant query returns after range query? Would explore be stuck in loading state?

Reply:
You are right it would.